actionview/lib/action_view/helpers/form_helper.rb

```diff
@@ -2610,7 +2610,7 @@ def fields_for_with_nested_attributes(association_name, association, options, bl
               else
                 options[:child_index] = nested_child_index(name)
               end
-              output << fields_for_nested_model("#{name}[#{options[:child_index]}]", child, options, block)
+              output << fields_for_nested_model("#{name}[#{options[:child_index]}]", child, options, block).to_s
```

Comment:
These `to_s` felt weird at first but they stem from the fact that `capture` can return a `nil` so this seems the best place to handle it 👍🏻

Reply:
Yes, I was hesitating between that and a `|| ""`. Maybe the later is better?

Reply:
Yes, I think that makes the intent clear

Reply:
Ok, I replaced them by actual `if ...`. No point causing a concatenation of an empty string.